test/DynamoCoreTests/SearchViewModelTests.cs

```diff
@@ -84,5 +87,381 @@ public void ShortenCategoryNameTests()
+
+            Assert.AreEqual(2, theClass.Items.Count);
+            Assert.IsNotNull(theClass.SubCategories.FirstOrDefault(c => c.Name == "SubCategory1"));
+            Assert.IsNotNull(theClass.SubCategories.FirstOrDefault(c => c.Name == "SubCategory2"));
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void RemoveEntry09RemoveAllNodes()
+        {
+            var searchElements = PrepareTestTree();
+
+            foreach (var element in searchElements)
+                viewModel.RemoveEntry(element.Model);
+
+            Assert.IsFalse(viewModel.BrowserRootCategories.Any());
+        }
+
+        #endregion
+
+        #region Helpers
+
+        private static NodeSearchElement CreateCustomNode(string name, string category,
+            string description = "", string path = "")
+        {
+            var element = new CustomNodeSearchElement(null,
+                new CustomNodeInfo(Guid.NewGuid(), name, category, description, path));
+
+            return element;
+        }
+
+        private static NodeSearchElementViewModel CreateCustomNodeViewModel(string name, string category,
+            string description = "", string path = "")
+        {
+            var element = CreateCustomNode(name, category, description, path);
+            return new NodeSearchElementViewModel(element);
+        }
+
+        private static NodeSearchElementViewModel CreateCustomNodeViewModel(NodeSearchElement element)
+        {
+            return new NodeSearchElementViewModel(element);
+        }
+
+        /// <summary>
+        /// Build complex tree.
+        /// </summary>
+        /// <returns>All members of built tree.</returns>
+        private static IEnumerable<NodeSearchElementViewModel> PrepareTestTree()
+        {
+            // Next tree will be built.
+            // Used blocks: ─, │, ┌, ┐, ┤, ├, ┴, ┬. 
+            //
+            //                                        Top
+            //                          ┌──────────────┼────────────┐
+            //                       Sub1_1         Sub1_2       Sub1_3
+            //             ┌────────────┤              │            │
+            //          Sub2_1         Classes      Sub2_3       Sub2_4
+            //    ┌────────┤            │              │            ├────────────┐
+            // Classes     Member2   Sub2_2         Classes         Member6   Classes
+            //    │                     │              │                         │
+            // Sub3_1                   Member3        │                      Sub3_4
+            //    │                              ┌─────┴─────┐                   |
+            //    Member1                     Sub3_2      Sub3_3                 Member7
+            //                                   │           │
+            //                                   Member4     Member5
+            //
+            var searchElements = new List<NodeSearchElementViewModel>();
```

Comment:
Thanks. I have used a part of this scheme for `RemoveEntry` piece of code explanation. Hope it will help.